index.js

```diff
@@ -48,7 +48,7 @@ const supportsAbortController = typeof globals.AbortController === 'function';
 const supportsStreams = typeof globals.ReadableStream === 'function';
 const supportsFormData = typeof globals.FormData === 'function';
 
-const mergeHeaders = (source1, source2) => {
+const mergeHeaders = (source1 = {}, source2 = {}) => {
```

Comment:
This would work with undefined perfectly with undefined, but should null be passed as an argument it will not get the default value. And as you wrote in https://github.com/sindresorhus/ky/issues/260#issuecomment-646725006 it will blow up in the Browser. So the best solution would properly be like suggested in https://github.com/sindresorhus/ky/issues/260#issuecomment-646718654 with a ternary check